client/src/boot/BootRoutes.js

```diff
@@ -6,7 +6,7 @@ import $ from 'jquery';
 import React from 'react';
 import { Provider as ReduxProvider } from 'react-redux';
 import ReactDOM from 'react-dom/client';
-import { BrowserRouter } from 'react-router-dom';
+import { createBrowserRouter, createRoutesFromElements, RouterProvider } from 'react-router-dom';
```

Comment:
We need to use `createBrowserRouter()` and `createRoutesFromElements()` instead of adding the `BrowserRouter` and `Routes` components ourselves because the `RouterProvider` doesn't like us doing it ourself - and we have to use the `RouterProvider` because this instantiates a "data router" which is necessary to use the new `useBlocker` hook.